src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java

```diff
@@ -0,0 +1,218 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.sscs.models.GapsEvent;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.CaseData;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Doc;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Documents;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Event;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Events;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Evidence;
+
+@RunWith(JUnitParamsRunner.class)
+public class CcdCasesSenderTest {
+
+    private static final String CASE_DETAILS_JSON = "src/test/resources/CaseDetailsWithOneEventAndNoEvidence.json";
+    private static final String CASE_DETAILS_WITH_ONE_EVIDENCE_AND_ONE_EVENT_JSON =
+        "src/test/resources/CaseDetailsWithOneEvidenceAndOneEvent.json";
+    @Mock
+    private UpdateCoreCaseDataService updateCoreCaseDataService;
+    @Mock
+    private CreateCoreCaseDataService createCoreCaseDataService;
+    @Mock
+    private SearchCoreCaseDataService searchCoreCaseDataService;
+    @Mock
+
+    private CcdCasesSender ccdCasesSender;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        ccdCasesSender = new CcdCasesSender(createCoreCaseDataService, searchCoreCaseDataService,
+            updateCoreCaseDataService);
+    }
+
+    @Test
+    @Parameters({"APPEAL_RECEIVED", "RESPONSE_RECEIVED", "HEARING_BOOKED", "HEARING_POSTPONED", "APPEAL_LAPSED",
+        "APPEAL_WITHDRAWN", "HEARING_ADJOURNED", "APPEAL_DORMANT"})
+    public void givenThereIsAnEventChange_shouldUpdateCcd(GapsEvent gapsEvent) throws Exception {
+        when(searchCoreCaseDataService.findCaseByCaseRef(anyString()))
+            .thenReturn(Collections.singletonList(getCaseDetails(CASE_DETAILS_JSON)));
+
+        ccdCasesSender.sendUpdateCcdCases(Collections.singletonList(buildUpdateCaseData(gapsEvent)));
+
+        verify(updateCoreCaseDataService, times(1))
+            .updateCase(eq(buildUpdateCaseData(gapsEvent)), anyLong(), eq(gapsEvent.getType()));
+    }
+
+    private CaseDetails getCaseDetails(String caseDetails) throws Exception {
+        String caseDetailsJson = FileUtils.readFileToString(new File(caseDetails),
+            StandardCharsets.UTF_8.name());
+        ObjectMapper mapper = Jackson2ObjectMapperBuilder.json().build();
+        return mapper.readerFor(CaseDetails.class).readValue(caseDetailsJson);
+    }
+
+    @Test
+    @Parameters({"APPEAL_RECEIVED", "RESPONSE_RECEIVED", "HEARING_BOOKED", "HEARING_POSTPONED", "APPEAL_LAPSED",
```

Comment:
Is it really necessary to test that there is no event change for every single event type? Surely once is good enough?

Reply:
yeah, right. Thanks